### PR TITLE
feature/hogaramaInfra

### PR DESCRIPTION
**Pull request by fhochleitner**
This PR contains all resources for OpenShift deployment, added documentation and helper skricpts to install/upgrade/remove hogarama or only certain components on the cluster

### CODE REVIEW

File: OCP-Infrastructure/helm/scripts/functions/helm_uninstall.active [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    local csvprometheus=$(oc get subscription prometheus-operator -n ${namespace} -o yaml | grep currentCSV | cut -d ":" -f2 | awk '{$1=$1;print}') && echo "found clusterserviceversion: ${csvprometheus}"
+    remove-default "${@}"
+    execute "oc delete clusterserviceversion ${csveap} -n ${namespace}"
+    execute "oc delete clusterserviceversion ${csvgrafana} -n ${namespace}"
+    execute "oc delete clusterserviceversion ${csvprometheus} -n ${namespace}"
+}
+readonly -f remove-hogarama-commons
+[ "$?" -eq "0" ] || return $?
+
+function remove-amq() {
```
Comment (ErhardSiegl) [inline OCP-Infrastructure/helm/scripts/functions/helm_uninstall.active:93]: Code smell: general utility functions should not have dependencies to individual components. This breaks encapsulation of the component.

File: OCP-Infrastructure/helm/scripts/hogarama.sh [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    fi
+
+    if [[ ! -f "${HOME}/.kube/config" ]]; then
+        export KUBECONFIG=/.kube/config
+    fi
+
+    ## CHECK LOGGED-IN STATUS ON CLUSTER
+    rc="$(oc whoami > /dev/null 2>&1  ;echo $?)"
+    if [[ rc -gt 0 ]]; then
+        echo "You are not logged in to the OpenShift Cluster, please login and try again"
```
Comment (ErhardSiegl) [inline OCP-Infrastructure/helm/scripts/hogarama.sh:167]: Errors should be written to stderr

File: docs/build/html/_sources/gettingstarted.rst.txt [hunk 3]
```diff
@@ -0,0 +1,84 @@
+Getting Started
```
Comment (ErhardSiegl) [inline docs/build/html/_sources/gettingstarted.rst.txt:1]: Why is this here and not in https://github.com/Gepardec/Hogarama.wiki.git ?

File: docs/build/html/gettingstarted.html [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  <div class="section" id="getting-started">
+<h1>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
+<div class="section" id="requirements">
+<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
+<ul class="simple">
+<li><p>OpenShift 4.0 or above</p></li>
+<li><p>Docker</p></li>
+<li><p>Our infrastructure secrets</p></li>
+</ul>
+<p>In order to run Hogarama you need access to an OpenShift 4.0 cluster or above and to be able to run docker on your workstation. In addition to that you need our infrastructure secrets. The deployment script checks if it can locate the secrets file on your workstation and if it can’t find it, it will tell you where to get it from.</p>
```
Comment (ErhardSiegl) [inline docs/build/html/gettingstarted.html:171]: Looks like generated code and shouldn't be  checked in.

File: OCP-Infrastructure/helm/amq/values.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+credentials:
+  amq_password: "amq_password"
+  amq_user: "amq_user"
```
[the remark this answers is not in the document]
Comment (ckaserer) [reply OCP-Infrastructure/helm/amq/values.yaml:4]: What's your train of thought for that assumption? You can't login? you checked the secret stored in gdrive and its not the same as before?

File: OCP-Infrastructure/helm/hogajama/templates/crd.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+kind: WildFlyServer
+metadata:
+  name: hogajama
+spec:
+  applicationImage: 'image-registry.openshift-image-registry.svc:5000/{{ .Release.Namespace }}/hogajama-run:latest'
+  env:
+    - name: KEYCLOAK_AUTH_SERVER_URL
+      value: {{ .Values.hogarama.hogajama.keycloak.server_url }}
+    - name: POSTGRESQL_HOST
+      value: helm-hogajama-postgresql
```
[the remark this answers is not in the document]
Comment (ckaserer) [reply OCP-Infrastructure/helm/hogajama/templates/crd.yaml:11]: should be removed, correct

File: docs/build/html/_sources/gettingstarted.rst.txt (hunk 3 above)
Comment (ckaserer) [reply docs/build/html/_sources/gettingstarted.rst.txt:1]: through this approach, we can version control our documentation the same way we do our code. in addition, we can have snapshot our documentation for multiple releases. e.g. the documentation checked in to release x fits for release x since those are bundled together. you can easily jump from one version to the other and get the corresponding documentation, even if the doc content changes completely in a year from now

Comment (ckaserer) [reply docs/build/html/_sources/gettingstarted.rst.txt:1]: in addition, we can utilize GitHub pages to serve the docs at gepardec.github.io which we can redirect to a gepardec.com subdomain if we wish to do so

File: docs/build/html/gettingstarted.html (hunk 4 above)
Comment (ckaserer) [reply docs/build/html/gettingstarted.html:171]: yes that is generated code. Due to the way GitHub pages work I am not aware of any other way to serve documentation as github pages in the same manner as readthedocs.io does it without directly utilizing readthedocs as a hosting provider.

File: docs/build/html/_sources/gettingstarted.rst.txt (hunk 3 above)
Comment (ckaserer) [reply docs/build/html/_sources/gettingstarted.rst.txt:1]: *available via the above-mentioned link only once it is merged to master and github pages are enabled for the repository. an example of how this would look like can be locally inspected  by opening docs/build/html/index.html in a browser of your choice

Comment (ErhardSiegl) [reply docs/build/html/_sources/gettingstarted.rst.txt:1]: There are always more than one way to do it, but if everyone use his own nobody finds anything anymore. Such decisions should be discussed and decided within the project.

File: OCP-Infrastructure/helm/amq/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ErhardSiegl) [reply OCP-Infrastructure/helm/amq/values.yaml:4]: I looked at https://github.com/Gepardec/Hogarama/blob/master/Templates/local_openshift/hogarama-amq.yaml#L120

[the remark this answers is not in the document]
Comment (fhochleitner) [reply OCP-Infrastructure/helm/amq/values.yaml:4]: the values in values.yaml are dummy-values for testing and templating. but @ckaserer has more information on this. 

the correct values which will be used are stored in our GDrive. there should also be documentation for this made by clemens as well

File: OCP-Infrastructure/helm/scripts/functions/helm_uninstall.active [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+function helm-uninstall () {
+    local resources=("${!1}")
+    local flag_dryrun=${FLAG_DRYRUN:-false}}
+    local flag_quiet=${FLAG_QUIET:-false}}
+    local secrets_file=${TOPLEVEL_DIR}/helm/secrets/values.yaml
+    local rc_overall=0
+
+
+
+    ## CHECK SPECIAL RESOURCES COMMANDS
```
[the remark this answers is not in the document]
Comment (fhochleitner) [reply OCP-Infrastructure/helm/scripts/functions/helm_uninstall.active:12]: should be 2 occurences because uninstall requires the order to be reversed but it should be extracted nonetheless

**Comment by fhochleitner** [comment]
PR merged so we don't run out of sync with master. PR not deleted yet so we have information from reviews for cleanup in the future.

i would also recommend a meeting where clemens and me will give an overview on how to setup infrastructure on aws and how our current setup works
